### PR TITLE
feat: Adds a few standard loaders

### PR DESCRIPTION
**Pull request by jbwdevries**
These we've seen used in multiple projects, and they are generic and require no dependencies. So we can add them here, making the live of our users a little easier.

Also tries to improve the type in the printed errors.

### CODE REVIEW

**Comment by jbwdevries** [comment]
@RonNabuurs Can you review?

File: load_environ_typed/loaders.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    """
+    try:
+        return load_bool(raw)
+    except ValueError:
+        return Path(raw)
+
+
+def load_binary_file_from_path(raw: str) -> bytes:
+    try:
+        with open(raw, 'rb') as fil:
```
Comment (RonNabuurs) [inline load_environ_typed/loaders.py:85]: Maybe I have a preference for pathlib over the builtin open. Maybe I've been using ruff too much lately. https://docs.astral.sh/ruff/rules/#flake8-use-pathlib-pth

File: load_environ_typed/loaders.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            return fil.read()
+    except FileNotFoundError:
+        raise ValueError(f'File not found: {raw}')
+    except IsADirectoryError:
+        raise ValueError(f'Unexpected directory: {raw}')
+    except PermissionError:
+        raise ValueError(f'No permissions to read: {raw}')
+
+
+def load_ascii_file_from_path(raw: str) -> str:
```
Comment (RonNabuurs) [inline load_environ_typed/loaders.py:95]: What's your opinion about load_string method with as parameter `ASCII` or `UTF-8`, then you can use partials in the loader.

Comment (RonNabuurs) [reply load_environ_typed/loaders.py:95]: This solution might be a bit more descriptive and easy to use though.

File: load_environ_typed/loaders.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
+from typing import List, Optional, Union
+
+from pathlib import Path
+
+from load_environ_typed import load_bool
+
```
Comment (RonNabuurs) [inline load_environ_typed/loaders.py:7]: Another method I use in multiple project is `load_string_list`, it's a simple loader that splits the environment value on a (configurable?) delimiter.

File: load_environ_typed/loaders.py (hunk 2 above)
Comment (jbwdevries) [reply load_environ_typed/loaders.py:95]: These are the two standard use cases for loading text files, making people with with partial seems extra work for nothing

File: load_environ_typed/loaders.py (hunk 1 above)
Comment (jbwdevries) [reply load_environ_typed/loaders.py:85]: It doesn't seem to add anything except another class instantiation. Especially the example on ruff seems pointless.  This project uses pycodestyle and pyflakes. I'll keep it in mind for future reference, though.

File: load_environ_typed/loaders.py (hunk 3 above)
Comment (jbwdevries) [reply load_environ_typed/loaders.py:7]: I added load_list_int and load_list_str for your convenience. It's using a CSV parser for maximum flexibility

File: load_environ_typed/loaders.py (hunk 1 above)
Comment (RonNabuurs) [reply load_environ_typed/loaders.py:85]: Seems fine by me. Maybe it would be nice to add some linting to a CI job, but that could be done in the future.

**Review by RonNabuurs — APPROVED** [review]

**Pull request merged**